test/test_base_service.py

```diff
@@ -154,6 +155,27 @@ def test_fail_http_config():
     with pytest.raises(TypeError):
         service.with_http_config(None)
 
+@responses.activate
+def test_cwd():
+    file_path = os.path.join(
+        os.path.dirname(__file__), '../resources/ibm-credentials.env')
+    # Try changing working directories to test getting creds from cwd
+    cwd = os.getcwd()
+    os.chdir(os.path.dirname(file_path))
+    iam_authenticator = get_authenticator_from_environment('ibm-watson')
+    os.chdir(cwd)
```

Comment:
I don't quite understand why you'd restore the original CWD immediately after loading up an authenticator but before you construct the service instance (which presumably would make a call to `BaseService.configure_service()`.   What scenario are you trying to test exactly?   What would make sense to me would be to "cd" to the directory containing the `ibm-credentials.env` file, then try to construct an instance of the service while passing in a service name of "ibm_watson".   This should load up an authenticator via external config AND configure the service instance using the properties contained in that file.

Reply:
I mistakenly thought that getting the authenticator was the only thing that needed to be in the current working directory so I moved my cleanup statements to immediately after that so they would be called even if the assert statements failed. I moved it to it's correct location and it's working now, thanks!